src/hotspot/share/runtime/abstract_vm_version.hpp

```diff
@@ -56,13 +79,42 @@ class Abstract_VM_Version: AllStatic {
 
   // CPU feature flags, can be affected by VM settings.
   static uint64_t _features;
+
   static const char* _features_string;
 
   static const char* _cpu_info_string;
 
   // Original CPU feature flags, not affected by VM settings.
   static uint64_t _cpu_features;
 
+  // CPU feature flags vector, can be affected by VM settings.
+  static VM_Features _vm_target_features;
+
+  // Original CPU feature flags vector, not affected by VM settings.
+  static VM_Features _cpu_target_features;
+
+  static void sync_cpu_features() {
+    memcpy(_cpu_target_features._features_vector, _vm_target_features._features_vector,
```

Comment:
Any particular reason to use `memcpy`/`memset` and not a loop over `_features_vector` array?

I believe once you define default and copy constructors for `VM_Features`, `sync_cpu_features()` and `clear_cpu_features()` won't be needed anymore.